pkg/bff/admin.go

```diff
@@ -1,15 +1,130 @@
 package bff
 
 import (
+	"context"
 	"fmt"
 	"net/http"
 
 	"github.com/gin-gonic/gin"
+	"github.com/rs/zerolog/log"
+
 	"github.com/trisacrypto/directory/pkg/bff/api/v1"
+	"github.com/trisacrypto/directory/pkg/bff/auth"
+	"github.com/trisacrypto/directory/pkg/gds/admin/v2"
+	"github.com/trisacrypto/directory/pkg/gds/models/v1"
 )
 
+// GetCertificates makes parallel calls to the admin services to get the certificate
+// information for both testnet and mainnet. If testnetID or mainnetID are empty
+// strings, this will simply return a nil response for the corresponding network so
+// the caller can distinguish between a non registration and an error.
+func (s *Server) GetCertificates(ctx context.Context, testnetID, mainnetID string) (testcerts *admin.ListCertificatesReply, maincerts *admin.ListCertificatesReply, err error) {
+	// Create the RPC which can do both testnet and mainnet calls
+	rpc := func(ctx context.Context, client admin.DirectoryAdministrationClient, network string) (rep interface{}, err error) {
+		var vaspID string
+		switch network {
+		case testnet:
+			vaspID = testnetID
+		case mainnet:
+			vaspID = mainnetID
+		default:
+			return nil, fmt.Errorf("unknown network: %s", network)
+		}
+
+		if vaspID == "" {
+			// The VASP is not registered for this network, so do not error and return
+			// nil
+			return nil, nil
+		}
+		return client.ListCertificates(ctx, vaspID)
+	}
+
+	// Perform the parallel requests
+	results, errs := s.ParallelAdminRequests(ctx, rpc, false)
+	if len(errs) != 2 || len(results) != 2 {
+		return nil, nil, fmt.Errorf("unexpected number of results from parallel requests: %d", len(results))
+	}
+
+	// Parse the results
+	var ok bool
+	if errs[0] != nil {
+		return nil, nil, errs[0]
+	}
```

Comment:
I think I wanted to be able to expose the error to the frontend so it's able to differentiate between errors and "no certificates" responses, but it does make sense to return partial results. I think the change I have to make is change the json tag to `omitempty` and change the response slices to pointers?

Reply:
That reasoning makes sense - perhaps we should ask @masskoder and @eschmier what they would prefer? E.g. if partial results are returned it may seem to the user that some of the certificates are "missing". However, if there is an error the front end can retry periodically. 

Slices are already pointers, so you don't have to make that change. If we do return partial, we'll likely want to include any errors in the response so that the front-end knows something went wrong with one of the directories.